### PR TITLE
ENH: Improve DataImporter

### PR DESCRIPTION
**Pull request by phcerdan**
Allows reading LabelMaps, Segmentations, and Models.

Things to improve / todo:
- Functions to display segments should be moved outside logic into widget
- Table might contain more information other than just the name, the interface is
not great.
- Think about re-implementing IO for a csv file (csv is used in other SALT extensions)

### CODE REVIEW

**Comment by phcerdan** [comment]
Closes #54

File: Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py [hunk 1]
```diff
@@ -17,8 +21,10 @@ def __init__(self, parent):
     self.parent.title = "Data Importer"
     self.parent.categories = ["Shape Analysis Toolbox"]
     self.parent.dependencies = []
-    self.parent.contributors = ["Hina Shah (Kitware Inc.)"]
+    self.parent.contributors = ["Hina Shah (Kitware Inc.), Pablo Hernandez (Kitware Inc,)"]
     self.parent.helpText = """
+    This module import label images and segmentations from files and folders and compute the topology number of each segment.
+    topologyNumber = cleanData.GetNumberOfPoints() - edges.GetNumberOfLines() + cleanData.GetNumberOfPolys()
```
Comment (jcfr) [inline Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py]: It this comment correct ?

File: Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py [hunk 2]
```diff
@@ -33,108 +39,324 @@ class DataImporterLogic(ScriptedLoadableModuleLogic):
...
+    self.labelMapDict = {}
+    self.modelDict = {}
     self.segmentationDict = {}
+    self.labelRangeInCohort = (-1, -1)
     self.topologyDict = {}
-    self.consistentTopologyDict = {}
     self.polyDataDict = {}
-
+    self.TOPOLOGY_TYPES = {
+        0: 'Circle/Torus/Mobius Strip',
```
Comment (jcfr) [inline Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py]: Introduce a constant for each one:

```
TOPOLOGY_STRIP = 0
TOPOLOGY_DISK = 1
...
```

or 

```
TOPOLOGY_STRIP_TYPE = 0
TOPOLOGY_DISK_TYPE = 1
...
```

then, you would use these constant in the code.

File: Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py [hunk 3]
```diff
@@ -33,108 +39,324 @@ class DataImporterLogic(ScriptedLoadableModuleLogic):
...
+    }
+    # help variable to map continious indices to TOPOLOGY_TYPES. Used in comboBoxes
+    self.topologyTypeToIndex = {
+        0: 0,
+        1: 1,
+        2: 2,
+        -2: 3,
+        -4: 4,
+    }
+    self.indexToTopologyType = {
```
Comment (jcfr) [inline Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py]:
```
indexToTopologyType = {index: type for type, index in self.topologyTypeToIndex.items()}
```

File: Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py [hunk 4]
```diff
@@ -33,108 +39,324 @@ class DataImporterLogic(ScriptedLoadableModuleLogic):
...
   def cleanup(self):
-    print 'Deleting nodes'
-    if self.testCaseDict is not None:
-      for node_name in self.testCaseDict.keys():
-        print 'Deleting node: ' + node_name
-        MRMLUtility.removeMRMLNode(self.testCaseDict[node_name])
-        MRMLUtility.removeMRMLNode(self.segmentationDict[node_name])
+    logging.debug('Deleting nodes')
+    if self.labelMapDict is not None:
+      for nodeName in self.labelMapDict.keys():
```
Comment (jcfr) [inline Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py]: fyi ..  using `.keys()` is optional

File: Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py [hunk 5]
```diff
@@ -33,108 +39,324 @@ class DataImporterLogic(ScriptedLoadableModuleLogic):
...
+      logging.warning('Segmentation in path: {} has not been loaded into segmentationDict.'.format(path))
+      return False
+
+    # Add to the dicts only if succesful
+    self.segmentationDict[fileName] = segmentationNode
+    self.labelRangeInCohort = labelRange
+    return True
+
+  def filePathsFromCSVFile(self, csvFileName):
+    """
```
Comment (jcfr) [inline Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py]: May be you could also use the Slicer reader for loading table file (aka csv)?

File: Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py [hunk 6]
```diff
@@ -33,108 +39,324 @@ class DataImporterLogic(ScriptedLoadableModuleLogic):
...
+      {'segmentName0': '0', 'segmentName1': '1'},
+    name1:
+      {'segmentName0': '1', 'segmentName1': '0'}
+    name2:
+      {'segmentName0': '1', 'segmentName1': '0'}
+    }
+    It would return '1' if inputSegmentName == 'segmentName0'
+                    '0' if inputSegmentName == 'segmentName1'
+    """
+    from collections import Counter
```
Comment (jcfr) [inline Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py]: import at the top ?

File: Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py [hunk 7]
```diff
@@ -33,108 +39,324 @@ class DataImporterLogic(ScriptedLoadableModuleLogic):
...
-  def getLabelRangeInCohort(self):
-    return self.labelRangeInCohort
+  def _computeModeOfSegment(self, inputTopologyDict, inputSegmentName):
+    """
+    Compute the mode of the segmentName among the population
+    Raise error if input dict is empty or not nested.
+    Returns the mode value, or
+    None if inputSegmentName is not found in the dictionary.
+
+    Example:
```
Comment (jcfr) [inline Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py]: Sphinx would be:

```
Example::
  {
    name0:
    [...]  
```

**Pull request merged**